extensions/objects/templates/svg-filename-editor.component.ts

```diff
@@ -425,79 +506,22 @@ angular.module('oppia').component('svgFilenameEditor', {
                 if (objId !== '') {
-                  if (objId.slice(0, 5) === 'group') {
-                    if (groupedObjects.length <= objId.slice(5)) {
-                      groupedObjects.push([]);
-                    }
-                    obj.toSVG = ctrl.createCustomToSVG(
-                      obj.toSVG, obj.type, obj.id);
-                    groupedObjects[objId.slice(5)].push(obj);
-                  }
+                  groupedObjects = loadGroupedObject(
+                    objId, obj, groupedObjects);
                 } else {
-                  if (obj.get('type') === 'rect') {
-                    if (
-                      elements[index].width.baseVal.valueAsString === '100%' &&
-                      elements[index].height.baseVal.valueAsString === '100%') {
-                      ctrl.canvas.setBackgroundColor(obj.get('fill'));
-                    } else {
-                      ctrl.canvas.add(obj);
-                    }
-                  } else if (obj.type === 'text') {
-                    var element = elements[index];
-                    var childrens = [].slice.call(element.childNodes);
-                    var value = '';
-                    var coloredTextIndex = [];
-                    childrens.forEach(function(el, index) {
-                      if (el.nodeName === 'tspan') {
-                        if (el.style.fill !== '') {
-                          coloredTextIndex.push({
-                            startIndex: value.length,
-                            endIndex: (
-                              value.length +
-                              el.childNodes[0].nodeValue.length),
-                            fill: el.style.fill,
-                            stroke: el.style.stroke,
-                            strokeWidth: el.style.strokeWidth
-                          });
-                        }
-                        value += el.childNodes[0].nodeValue;
-                      }
-                      if (
-                        index < childrens.length - 1 &&
-                        el.style.fill === '') {
-                        value += '\n';
-                      }
-                    });
-                    value = (
-                      obj['text-transform'] === 'uppercase' ?
-                      value.toUpperCase() : value);
-
-                    obj.set({
-                      text: value,
-                    });
-                    var text = new fabric.Textbox(obj.text, obj.toObject());
-                    text.set({
-                      type: 'textbox',
-                      strokeUniform: true,
+                  // This if condition is to detect the background color
+                  // which has a rect shape.
+                  if (
+                    obj.get('type') === 'rect' &&
+                    elements[index].width.baseVal.valueAsString === '100%' &&
+                    elements[index].height.baseVal.valueAsString === '100%') {
```

Comment:
Instead of a comment, can you extract the condition into a boolean function with a higher-level name, e.g. isFullRectangle() or something similar?